components/UserObserver.php

```diff
@@ -100,29 +205,37 @@ public function is_stale($username)
 		return (strtotime($user->last_modified_date) < (time() - Yii::app()->params['mehstaffdb_cache_time']));
 	}
 
-	private function getDoctorGradeFromJobTitle($jobTitle){
-		$MEHDescription = array(
-			"1" => "Consultant",
-			"3" => "Associate Specialist",
-			"4" => "Fellow",
-			"5" => "Specialist Registrar",
-			"7" => "Trust Doctor",
-			"8" => "Senior House Officer",
-			"16" => "House Officer",
-			"20" => "Anaesthetist",
-			"21" => "Orthoptist",
-			"22" => "Optometrist",
-			"23" => "Clinical nurse specialist",
-			"24" => "Nurse",
-			"25" => "Health Care Assistant",
-			"26" => "Ophthalmic Technician",
-			"27" => "Surgical Care Practitioner",
-			"28" => "Clinical Assistant",
-			"29" => "RG1",
-			"30" => "RG2",
-			"31" => "ODP",
-			"32" => "Administration staff");
-		//"33" => "Other"
+	/**
+     * Connect job title to doctor grade id
+     *
+     * @param string $jobTitle
+     * @return int ID of the doctor grade
+     */
+	private function getDoctorGradeFromJobTitle(string $jobTitle): int
+	{
+		$MEHDescription = array
```

Comment:
if the jobTitle passed in is dynamically fetched from remote, should this list be fetched from remote as well? Ignore me if it is not achievable

Reply:
This is how it worked in V1.19 and I kept the original version.

Reply:
Well, but it still feels more robust if the list is fetched from the remote db.

Reply:
The remote can't provide such a list.